src/main/java/seedu/weme/model/statistics/DislikeData.java

```diff
@@ -93,4 +93,28 @@ public void deleteDislikesByMemeRef(String memeRef) {
         }
         return copy;
     }
+
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof DislikeData)) {
+            return false;
+        }
+
+        DislikeData otherDislikeData = (DislikeData) other;
+        if (!dislikeMap.keySet().equals(otherDislikeData.getCopy().keySet())) {
+            return false;
+        } else {
+            for (Map.Entry<String, SimpleIntegerProperty> entry : otherDislikeData.dislikeMap.entrySet()) {
```

Comment:
I was thinking that you could import java.util.Map.Entry. And maybe simplify this one liner statement. Assign a variable to get the entrySet() first then iterate through this variable.

Reply:
Line 92 also.